src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/HotReload/ProjectHotReloadSession.cs

```diff
@@ -17,46 +20,67 @@ internal sealed class ProjectHotReloadSession : IProjectHotReloadSessionInternal
     private readonly Lazy<IHotReloadAgentManagerClient> _hotReloadAgentManagerClient;
     private readonly ConfiguredProject _configuredProject;
     private readonly Lazy<IHotReloadDiagnosticOutputService> _hotReloadOutputService;
-    private readonly Lazy<IManagedDeltaApplierCreator> _deltaApplierCreator;
     private readonly IProjectHotReloadSessionCallback _callback;
     private readonly IProjectHotReloadBuildManager _buildManager;
     private readonly ILaunchProfile _launchProfile;
     private readonly DebugLaunchOptions _debugLaunchOptions;
+    private readonly IHotReloadDebugStateProvider _debugStateProvider;
     private readonly IProjectHotReloadLaunchProvider _launchProvider;
 
     private bool _sessionActive;
-    private IDeltaApplier? _deltaApplier;
+    private IDeltaApplier? _lazyDeltaApplier;
 
     public ProjectHotReloadSession(
         string name,
         int id,
         Lazy<IHotReloadAgentManagerClient> hotReloadAgentManagerClient,
         Lazy<IHotReloadDiagnosticOutputService> hotReloadOutputService,
-        Lazy<IManagedDeltaApplierCreator> deltaApplierCreator,
         IProjectHotReloadSessionCallback callback,
         IProjectHotReloadBuildManager buildManager,
         IProjectHotReloadLaunchProvider launchProvider,
         ConfiguredProject configuredProject,
         ILaunchProfile launchProfile,
-        DebugLaunchOptions debugLaunchOptions)
+        DebugLaunchOptions debugLaunchOptions,
+        IHotReloadDebugStateProvider debugStateProvider)
     {
         Name = name;
         Id = id;
 
         _configuredProject = configuredProject;
         _hotReloadAgentManagerClient = hotReloadAgentManagerClient;
         _hotReloadOutputService = hotReloadOutputService;
-        _deltaApplierCreator = deltaApplierCreator;
         _callback = callback;
         _launchProfile = launchProfile;
         _buildManager = buildManager;
         _launchProvider = launchProvider;
         _debugLaunchOptions = debugLaunchOptions;
+        _debugStateProvider = debugStateProvider;
     }
 
-    // IProjectHotReloadSession
+    /// <summary>
+    /// Returns true and the path to the client agent implementation binary if the application needs the agent to be injected.
+    /// </summary>
+    private static string GetStartupHookPath(Version applicationTargetFrameworkVersion)
+    {
+        var hookTargetFramework = applicationTargetFrameworkVersion.Major >= 10 ? "net10.0" : "net6.0";
```

Comment:
Under what circumstances would these hard coded values need to be updated? Could a comment here be helpful?

Reply:
Added comment to `Microsoft.VisualStudio.ProjectSystem.Managed.csproj`, where the dll is referenced and copied to the output dir.